src/Catalog/Helpers/FeedHelpers.cs

```diff
@@ -355,6 +308,78 @@ private static DateTime ForceUtc(DateTime date)
+            var packageUri = Utilities.GetNugetCacheBustingUri(feedPackageDetails.ContentUri, workItem.Timestamp.ToString("O"));
+            HttpResponseMessage response = null;
+            try
+            {
+                using (telemetryService.TrackDuration(
+                    TelemetryConstants.PackageDownloadSeconds,
+                    new Dictionary<string, string>()
+                    {
+                        { TelemetryConstants.Id, feedPackageDetails.PackageId?.ToLowerInvariant() },
+                        { TelemetryConstants.Version, feedPackageDetails.PackageVersion?.ToLowerInvariant() },
+                    }))
+                {
+                    response = await client.GetAsync(packageUri, cancellationToken);
+                }
+            }
+            catch (TaskCanceledException tce)
+            {
+                // If the HTTP request timed out, a TaskCanceledException will be thrown.
+                throw new HttpClientTimeoutException($"HttpClient request timed out in {nameof(FeedHelpers.DownloadMetadata2CatalogAsync)}.", tce);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                using (var stream = await response.Content.ReadAsStreamAsync())
+                {
+                    workItem.PackageCatalogItem = Utils.CreateCatalogItem(
+                        feedPackageDetails.ContentUri.ToString(),
+                        stream,
+                        feedPackageDetails.CreatedDate,
+                        feedPackageDetails.LastEditedDate,
+                        feedPackageDetails.PublishedDate);
+
+                    if (workItem.PackageCatalogItem != null)
+                    {
+                        logger?.LogInformation("Add metadata from: {PackageDetailsContentUri}", feedPackageDetails.ContentUri);
+                    }
+                    else
+                    {
+                        logger?.LogWarning("Unable to extract metadata from: {PackageDetailsContentUri}", feedPackageDetails.ContentUri);
+                    }
+                }
+            }
+            else
+            {
+                if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    //  the feed is out of sync with the actual package storage - if we don't have the package there is nothing to be done we might as well move onto the next package
+                    logger?.LogWarning("Unable to download: {PackageDetailsContentUri}. Http status: {HttpStatusCode}", feedPackageDetails.ContentUri, response.StatusCode);
+                }
+                else
+                {
+                    //  this should trigger a restart - of this program - and not move the cursor forward
+                    logger?.LogError("Unable to download: {PackageDetailsContentUri}. Http status: {HttpStatusCode}", feedPackageDetails.ContentUri, response.StatusCode);
+                    throw new Exception(
```

Comment:
How big is the batch of packages given to `DownloadMetadata2CatalogAsync`? If the batch is big, this behavior of throwing without retrying may be expensive.